java/client/src/main/java/com/vesoft/nebula/client/graph/meta/MetaClient.java

```diff
@@ -0,0 +1,561 @@
+            Map<Long, EdgeItem> edges = Maps.newHashMap();
+            Map<String, Long> edgesName = Maps.newHashMap();
+            Map<Long, String> edgesId = Maps.newHashMap();
+            for (EdgeItem item : listEdges(spaceName)) {
+                edges.put((long) item.getEdge_type(), item);
+                edgesName.put(new String(item.edge_name), (long) item.getEdge_type());
+                edgesId.put((long) item.getEdge_type(), new String(item.edge_name));
+            }
+            metaInfo.getSpaceEdgeItems().put(spaceName, edges);
+            metaInfo.getEdgeNameMap().put(spaceName, edgesName);
+            metaInfo.getEdgeIdMap().put(spaceName, edgesId);
+        }
+        return this;
+    }
+
+
+    /**
+     * get all spaces info
+     *
+     * @return empty list if exception happen
+     */
+    private List<IdName> listSpaces() {
+        ListSpacesReq request = new ListSpacesReq();
+        ListSpacesResp response;
+        try {
+            response = client.listSpaces(request);
+        } catch (TException e) {
+            LOGGER.error(String.format("List Spaces Error: %s", e.getMessage()));
+            return Lists.newLinkedList();
+        }
+        if (response.getCode() == ErrorCode.SUCCEEDED) {
+            return response.getSpaces();
+        } else {
+            LOGGER.error(String.format("List Spaces Error Code: %d", response.getCode()));
+            return Lists.newLinkedList();
+        }
+    }
+
+
+    /**
+     * get all parts and part's location
+     *
+     * @param spaceName nebula graph space
+     * @return empty map if exception happen
+     */
+    private Map<Integer, List<HostAndPort>> getPartsLocation(String spaceName) {
+        Map<Integer, List<HostAndPort>> result;
+        try {
+            result = getPartsAlloc(spaceName);
+        } catch (ExecuteFailedException | TException e) {
+            return Maps.newHashMap();
+        }
+        return result;
+    }
+
+    /**
+     * get all tags info
+     *
+     * @param spaceName nebula graph space
+     * @return empty list if exception happen
+     */
+    private List<TagItem> listTags(String spaceName) {
+        List<TagItem> tagItems;
+        try {
+            tagItems = getTags(spaceName);
+        } catch (TException | ExecuteFailedException e) {
+            return Lists.newLinkedList();
+        }
+        return tagItems;
+    }
+
+    /**
+     * get all edges info
+     *
+     * @param spaceName nebula graph space
+     * @return empty list if exception happen
+     */
+    private List<EdgeItem> listEdges(String spaceName) {
+        List<EdgeItem> edgeItems;
+        try {
+            edgeItems = getEdges(spaceName);
+        } catch (TException | ExecuteFailedException e) {
+            return Lists.newLinkedList();
+        }
+        return edgeItems;
+    }
+
+    /**
+     * check if space exist
+     */
+    private boolean existSpace(String spaceName) {
+        List<IdName> spaces = listSpaces();
```

Comment:
existSpace check  data from cache, not directly to get from metad.

Reply:
the method is called only when cache doesn't contain space.